.pipelines/cni/pipeline.yaml

```diff
@@ -391,6 +403,125 @@ stages:
       arch: amd64
       dualstackVersion: ${CILIUM_DUALSTACK_VERSION}
       cni: "cilium_dualstack"
+      upgradeScenario: false
+
+
+  ## If upgradeScenario is set, redeploy new IPAM version to existing clusters and run tests
+  - ${{if eq(parameters.upgradeScenario, true)}}:

```

Comment:
if this is performing the same e2e test as above, would it be possible to make a template that performs the upgrade test where we pass all these parameters and the template automatically creates the before and after upgrade stages? The template could potentially also take in the upgrade scenario as true or false or a pipeline variable/expression for enable/disable. If we add more scenarios in the future I feel like it would be easy to forget to add the upgrade stage

Reply:
will keep this in mind to address in a follow up